libafl/src/fuzzer/mod.rs

```diff
@@ -354,25 +391,66 @@ where
             res = ExecuteInputResult::Solution;
         } else {
             #[cfg(not(feature = "introspection"))]
-            let is_corpus = self
+            let corpus_worthy = self
                 .feedback_mut()
-                .is_interesting(state, manager, &input, observers, exit_kind)?;
+                .is_interesting(state, manager, input, observers, exit_kind)?;
 
             #[cfg(feature = "introspection")]
-            let is_corpus = self
+            let corpus_worthy = self
                 .feedback_mut()
                 .is_interesting_introspection(state, manager, &input, observers, exit_kind)?;
 
-            if is_corpus {
+            if corpus_worthy {
                 res = ExecuteInputResult::Corpus;
             }
         }
+        Ok(res)
+    }
 
-        match res {
+    fn execute_and_process<EM>(
+        &mut self,
+        state: &mut Self::State,
+        manager: &mut EM,
+        input: <Self::State as UsesInput>::Input,
+        observers: &OT,
+        exit_kind: &ExitKind,
+        send_events: bool,
+    ) -> Result<(ExecuteInputResult, Option<CorpusId>), Error>
+    where
+        EM: EventFirer<State = Self::State>,
+    {
+        let exec_res = self.execute_with_res(state, manager, &input, observers, exit_kind)?;
```

Comment:
what is the reason to separate the previous process_execution into two function?

Reply:
ah ok. you had to do this because you want process_execution not to add testcases.
imo you should keep what process_execution does.

instead of changing every occurence of `process_execution` to `execute_and_process`, 
can you keep process_execution the same, 
but just make another method evaluate_without_generating_testcase that calls into execute_with_res. 

and from state you use this function

Reply:
I don't think that's the best approach as "process execution" implies processing the result of execution. I was quite confused when the function did both, execute and process. IMO this is a more semantic API even though it may break existing code.

Reply:
ok. 
but can you rename `execute_with_res` to something else? it sounds like execute_with_res takes an argument `res`